react-with-json-logic/src/testCaseRunner/testCaseRunner.ts

```diff
@@ -0,0 +1,139 @@
+import { QuestionnaireEngine, Result } from "covquestions-js";
+import { Questionnaire, TestCase } from "covquestions-js/models/Questionnaire.generated";
+
+type TestResultSuccess = {
+  description: string;
+  success: true;
+};
+
+type TestResultError = {
+  description: string;
+  success: false;
+  errorMessage: string;
+};
+
+export type TestResult = TestResultSuccess | TestResultError;
+
+export function runTestCases(testQuestionnaire: Questionnaire): TestResult[] {
+  if (testQuestionnaire.testCases === undefined || testQuestionnaire.testCases.length === 0) {
+    return [];
+  }
+
+  return testQuestionnaire.testCases.map((testCase) => runOneTestCase(testQuestionnaire, testCase));
+}
+
+export function runOneTestCase(testQuestionnaire: Questionnaire, testCase: TestCase): TestResult {
+  const timeOfExecution = testCase.options?.fillInDate ? Date.parse(testCase.options?.fillInDate) / 1000 : undefined;
+
+  const engine = new QuestionnaireEngine(testQuestionnaire, timeOfExecution);
+
+  const questionCheck = checkQuestions(engine, testCase);
+  if (questionCheck) {
+    return questionCheck;
+  }
+
+  const resultsCheck = checkResults(engine.getResults(), testCase);
+  if (resultsCheck) {
+    return resultsCheck;
+  }
+
+  return { description: testCase.description, success: true };
+}
+
+function findUnusedElements(arrayWithPossibleUnusedElements: string[], otherArray: string[]): string[] {
```

Comment:
Can we add a comment that "otherArray" needs to be a subset of "arrayWithPossibleUnusedElements"?

It's quite a minor issue, but something to stumble over when maintaining the code later.

Reply:
I was also struggling a long time with this function and the naming and I am still not satisfied... I will add the comment.

Reply:
I renamed the second argument to subSetOfFirstArray